File: MixedToImproper.py
# No doubt this is a module

# A class
class Mixed2Improper:
    
# A constructor
    def __init__(self, fraction):
        self.whole_number = fraction['whole_number']
        self.numerator    = fraction['numerator']
        self.denominator  = fraction['denominator']
        

    # Returns a scalar of the new numerator
    def doConvert(self):
		# STEPS 1, 2:
        return self.whole_number * self.denominator + self.numerator

File: MixedToImproper.vb
Public Class MixedToImproper

    Dim whole_number As Integer
    Dim numerator As Integer
    Dim denominator As Integer

    ' A constructor
    Public Sub _init_(fraction As Dictionary(Of String, Integer))
        whole_number = fraction.Item("whole_number")
        numerator = fraction.Item("numerator")
        denominator = fraction.Item("denominator")
    End Sub


    ' Returns a scalar Of the New numerator
    Public Function doConvert() As Integer
        ' STEPS 1, 2:
        Return whole_number * denominator + numerator
    End Function

End Class
